core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -19,138 +19,357 @@
+    public boolean lowercaseExpandedTerms() {
+        return this.settings.lowercaseExpandedTerms();
+    }
+
+    /** Specifies the locale for parsing terms. Defaults to ROOT if none is set. */
     public SimpleQueryStringBuilder locale(Locale locale) {
-        this.locale = locale;
+        this.settings.locale(locale);
         return this;
     }
 
+    /** Returns the locale for parsing terms for this query. */
+    public Locale locale() {
+        return this.settings.locale();
+    }
+
+    /** Specifies whether query parsing should be lenient. Defaults to false. */
     public SimpleQueryStringBuilder lenient(boolean lenient) {
-        this.lenient = lenient;
+        this.settings.lenient(lenient);
         return this;
     }
 
+    /** Returns whether query parsing should be lenient. */
+    public boolean lenient() {
+        return this.settings.lenient();
+    }
+
+    /** Specifies whether wildcards should be analyzed. Defaults to false. */
     public SimpleQueryStringBuilder analyzeWildcard(boolean analyzeWildcard) {
-        this.analyzeWildcard = analyzeWildcard;
+        this.settings.analyzeWildcard(analyzeWildcard);
         return this;
     }
 
+    /** Returns whether wildcards should by analyzed. */
+    public boolean analyzeWildcard() {
+        return this.settings.analyzeWildcard();
+    }
+
+    /**
+     * Specifies the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
+    /**
+     * Returns the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
+    public String minimumShouldMatch() {
+        return minimumShouldMatch;
+    }
+
+    /** Returns an exception if any mandatory fields are null. */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+
+        // Query text is required
+        if (queryText == null) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] query text missing",
+                    validationException);
+        }
+
+        // The following options should never be set to null or empty array.
+        // They are initialized with sane values, should the downstream user
+        // set them to something that doesn't make sense, complain here.
+        if (settings.locale() == null) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] locale missing",
+                    validationException);
+        }
+        if (flags == INVALID_FLAGS) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] flags missing",
```

Comment:
sorry I re-read t, missing is appropriate, maybe we could rename the constant to MISSING_FLAGS then